cg/meta/rsync/rsync_api.py

```diff
@@ -0,0 +1,189 @@
+
+    @property
+    def trailblazer_config_path(self) -> Path:
+        """Return Path to trailblazer config"""
+        return self.log_dir / "slurm_job_ids.yaml"
+
+    @property
+    def rsync_processes(self) -> Iterable[Path]:
+        """Yield existing rsync processes"""
+        yield from self.base_path.iterdir()
+
+    @staticmethod
+    def format_covid_destination_path(covid_destination_path: str, customer_id: str) -> str:
+        """Return destination path of covid report"""
+        return covid_destination_path % customer_id
+
+    @staticmethod
+    def get_trailblazer_config(slurm_job_id: int) -> Dict[str, List[str]]:
+        """Return dictionary of slurm job IDs"""
+        return {"jobs": [str(slurm_job_id)]}
+
+    @staticmethod
+    def write_trailblazer_config(content: dict, config_path: Path) -> None:
+        """Write slurm job IDs to a .YAML file used as the trailblazer config"""
+        LOG.info(f"Writing slurm jobs to {config_path.as_posix()}")
+        with config_path.open("w") as yaml_file:
+            yaml.safe_dump(content, yaml_file, indent=4, explicit_start=True)
+
+    @staticmethod
+    def process_ready_to_clean(before: dt.datetime, process: Path) -> bool:
+        """Return True if analysis is old enough to be cleaned"""
+
+        ctime: dt.datetime = dt.datetime.fromtimestamp(process.stat().st_ctime)
+
+        return before > ctime
+
+    def set_log_dir(self, ticket_id: int) -> None:
+        if self.log_dir.as_posix() == self.base_path.as_posix():
+            timestamp = dt.datetime.now()
+            timestamp_str = timestamp.strftime("%y%m%d_%H_%M_%S_%f")
+            folder_name = Path("_".join([str(ticket_id), timestamp_str]))
+            LOG.info(f"Setting log dir to: {self.base_path / folder_name}")
+            self.log_dir: Path = self.base_path / folder_name
+
+    def get_all_cases_from_ticket(self, ticket_id: int) -> List[models.Family]:
+        cases: List[models.Family] = self.status_db.get_cases_from_ticket(ticket_id=ticket_id).all()
+        return cases
+
+    def get_source_path(self, ticket_id: int) -> str:
+        cases: List[models.Family] = self.get_all_cases_from_ticket(ticket_id=ticket_id)
+        if not cases:
+            LOG.warning("Could not find any cases for ticket_id %s", ticket_id)
+            raise CgError()
+        customer_id: str = cases[0].customer.internal_id
+        delivery_source_path: str = (
+            Path(self.delivery_path, customer_id, "inbox", str(ticket_id)).as_posix() + "/"
+        )
+        return delivery_source_path
+
+    def get_destination_path(self, ticket_id: int) -> str:
+        cases: List[models.Family] = self.get_all_cases_from_ticket(ticket_id=ticket_id)
+        customer_id: str = cases[0].customer.internal_id
+        rsync_destination_path: str = (
+            Path(self.destination_path, customer_id, "inbox", str(ticket_id)).as_posix() + "/"
+        )
+        return rsync_destination_path
```

Comment:
I know you just moved this code so this one is optional.
The 2 functions: get_source_path and get_destination_path could be merge if you provide a path option. They are almost identical and I think it would be cleaner to just have one and you would see on the parameter call what to expect in return